src/test/java/seedu/address/logic/commands/RelateCommandTest.java

```diff
@@ -71,6 +71,32 @@ public void execute_invalidSecondPersonId_throwsCommandException() {
         assertThrows(CommandException.class, () -> command.execute(model));
     }
 
+    @Test
+    public void execute_samePersonId_throwsCommandException() {
+        Person person = new ClientBuilder().withId(ID_FIRST_PERSON.value).build();
+        model.addPerson(person);
+
+        IdContainsDigitsPredicate predicate = new IdContainsDigitsPredicate(List.of(ID_FIRST_PERSON.value,
+                ID_FIRST_PERSON.value));
+        RelateCommand command = new RelateCommand(predicate);
+        assertThrows(CommandException.class, () -> command.execute(model));
```

Comment:
Could possible use `assertThrows(Throwable, Executable, String)` or `assertCommandFailure(...)` to also verify the error message displayed to user.